Mathlib/LinearAlgebra/Matrix/HermitianFunctionalCalculus.lean

```diff
@@ -0,0 +1,302 @@
+/-
+Copyright (c) 2024 Jon Bannon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Bannon, Jireh Loreaux
+-/
+
+import Mathlib.LinearAlgebra.Matrix.Spectrum
+import Mathlib.Analysis.NormedSpace.Star.ContinuousFunctionalCalculus
+import Mathlib.Topology.ContinuousFunction.UniqueCFC
+import Mathlib.Analysis.NormedSpace.Star.Matrix
+
+/-!
+# Continuous Functional Calculus for Hermitian Matrices
+
+This file defines an instance of the continuous functional calculus for Hermitian matrices over an
+RCLike field 𝕜.
+
+## Main Results
+
+- definition of φ : the real StarAlgHom from C(spectrum ℝ A, ℝ) to (Matrix n n 𝕜) appearing in the
+                    instance.
+- instContinuousFunctionalCalculus : Instance of the Continuous functional Calculus for a hermitian
+                                     matrix A over 𝕜.
+
+## Tags
+
+spectral theorem, diagonalization theorem, continuous functional calculus
+-/
+
+section ConjugateUnits
+
+variable {R A : Type*} [CommSemiring R] [Ring A] [Algebra R A]
+
+/-- Conjugation by a unit preserves the spectrum, inverse on right. -/
+@[simp]
+lemma spectrum.conjugate_units {a : A} {u : Aˣ} :
+    spectrum R (u * a * u⁻¹) = spectrum R a := by
+  suffices ∀ (b : A) (v : Aˣ), spectrum R (v * b * v⁻¹) ⊆ spectrum R b by
+    refine le_antisymm (this a u) ?_
+    apply le_of_eq_of_le ?_ <| this (u * a * u⁻¹) u⁻¹
+    simp [mul_assoc]
+  intro a u μ hμ
+  rw [spectrum.mem_iff] at hμ ⊢
+  contrapose! hμ
+  simpa [mul_sub, sub_mul, Algebra.right_comm] using u.isUnit.mul hμ |>.mul u⁻¹.isUnit
+
+/-- Conjugation by a unit preserves the spectrum, inverse on left. -/
+@[simp]
+lemma spectrum.conjugate_units' {a : A} {u : Aˣ} :
+    spectrum R (u⁻¹ * a * u) = spectrum R a := by
+  simpa using spectrum.conjugate_units (u := u⁻¹)
+
+end ConjugateUnits
+
+section UnitaryConjugate
+
+universe u
+
+variable {R A : Type*} [CommSemiring R] [Ring A] [Algebra R A] [StarMul A]
+
+/-- Unitary conjugation preserves the spectrum, star on left. -/
+@[simp]
+lemma spectrum.unitary_conjugate {a : A} {u : unitary A} :
+    spectrum R (u * a * (star u : A)) = spectrum R a :=
+  spectrum.conjugate_units (u := unitary.toUnits u)
+
+/-- Unitary conjugation preserves the spectrum, star on right. -/
+@[simp]
+lemma spectrum.unitary_conjugate' {a : A} {u : unitary A} :
+    spectrum R ((star u : A) * a * u) = spectrum R a := by
+  simpa using spectrum.unitary_conjugate (u := star u)
+
+end UnitaryConjugate
+
+section FiniteSpectrum
```

Comment:
This section as should go in the same file as the theorems used to prove these, preferably in another PR.